dvclive/live.py

```diff
@@ -134,28 +144,51 @@ def next_step(self):
         self.set_step(self.get_step() + 1)
 
     def log(self, name: str, val: Union[int, float]):
-        data = None
-        if name in self._data:
-            data = self._data[name]
-        else:
-            for data_type in DATA_TYPES:
-                if data_type.could_log(val):
-                    data = data_type(name, self.dir)
-                    self._data[name] = data
-        if data is None:
+        if not Scalar.could_log(val):
             raise InvalidDataTypeError(name, type(val))
 
+        if name in self._scalars:
+            data = self._scalars[name]
+        else:
+            data = Scalar(name, self.dir)
+            self._scalars[name] = data
+
         data.dump(val, self._step)
 
         if self._summary:
             self.make_summary()
 
+    def log_image(self, name: str, val):
+        if not Image.could_log(val):
+            raise InvalidDataTypeError(name, type(val))
```

Comment:
Maybe `val` should be renamed. `val` made sense for scalars (`log()`) but `image_data`** or something more descriptive would be better here IMO.